### PR TITLE
Do not let git use a pager

### LINKED ISSUE #94988: Git output with user defined pager
- VSCode Version: 1.44.0
- OS Version: Windows 10 / 1909

### The issue
Git allows user to set up a custom pager app. See [`core.pager`](https://www.git-scm.com/docs/git-config#Documentation/git-config.txt-corepager) option in git configuration.
And this can ruin VS Code communication with git. Commands such as `git show`/`git diff`/etc., which are uses pager, can truncate output or provide invalid output.

### How to fix it
VS Code just need to pass [`--no-pager`](https://www.git-scm.com/docs/git/2.26.0#Documentation/git.txt---no-pager) arg when calling git. To always receive the raw output.
For example here:
https://github.com/HanabishiRecca/vscode/commit/a13569c2aa886c831fdcd8b13cf4ea3ccc3d475a

### PR DESCRIPTION
**Pull request by avdg**
This PR fixes #94988

### CODE REVIEW

**Comment by avdg** [comment]
As I just cloned the codebase, I was not yet able to create a build just yet (I do have already VS 2019 installed but I am not sure if I have to install 2017 as well)

File: extensions/git/src/main.ts [hunk 1]
```diff
@@ -36,7 +36,9 @@ async function createModel(context: ExtensionContext, outputChannel: OutputChann
 	const pathHint = workspace.getConfiguration('git').get<string>('path');
 	const info = await findGit(pathHint, path => outputChannel.appendLine(localize('looking', "Looking for git in: {0}", path)));
 
-	let env: any = {};
+	let env: any = {
+		GIT_PAGER: ''
+	};
```
Comment (joaomoreno) [inline extensions/git/src/main.ts:41]: Let's just do this here instead: https://github.com/microsoft/vscode/blob/95f26d2005ea0633aa48832d11e1f37068c4dcdb/extensions/git/src/git.ts#L541

**Review by joaomoreno — CHANGES_REQUESTED** [review]

File: extensions/git/src/git.ts [hunk 2]
```diff
@@ -540,7 +540,8 @@ export class Git {
 		options.env = assign({}, process.env, this.env, options.env || {}, {
 			VSCODE_GIT_COMMAND: args[0],
 			LC_ALL: 'en_US.UTF-8',
-			LANG: 'en_US.UTF-8'
+			LANG: 'en_US.UTF-8',
+			GIT_PAGER: ''
```
Comment (joaomoreno) [inline extensions/git/src/git.ts:544]: Are we sure the empty string does the trick?

> To disable pagination for all commands, set core.pager or GIT_PAGER to cat.

From https://git-scm.com/docs/git-config/2.17.0#Documentation/git-config.txt-pagerltcmdgt

Comment (avdg) [reply extensions/git/src/git.ts:544]: Probably better to follow recommendation

Comment (avdg) [reply extensions/git/src/git.ts:544]: I wonder if it works fine in powershell though...

Comment (avdg) [reply extensions/git/src/git.ts:544]: Seems to work fine

**Comment by joaomoreno** [comment]
Thanks!

**Pull request merged**